### PR TITLE
PHP 8 Support

### PR DESCRIPTION
**Pull request by adlenton**
Hey,

I took the liberty to propose a minimal changeset for php 8 support and with that also drop support for unsupported php versions (https://www.php.net/supported-versions.php).

- dropped support for php versions older than 7.2
- updated phpunit version
- compatibility fixes

### CODE REVIEW

**Comment by brainexe** [comment]
@serbanghita I'm also waiting for this PR, as it's blocking us from a full PHP8 upgrade :)

**Comment by serbanghita** [comment]
@adlenton Thank you, I actually needed this! I'm review-ing it and this will be the basis of `3.0.0`

File: tests/BasicsTest.php [hunk 1]
```diff
@@ -241,39 +241,6 @@ public function testSetLongUserAgent() {
         $this->assertEquals(strlen($md->getUserAgent()), 500);
     }
 
-    public function testSetDetectionType()
```
Comment (serbanghita) [inline tests/BasicsTest.php:244]: Why was this removed? At the moment, just looking at the code I don't remember the actual functionality and purpose.

File: .travis.yml [hunk 2]
```diff
@@ -4,14 +4,9 @@ language: php
...
-    - php: 5.3
-      dist: precise
-    - php: 5.4
-    - php: 5.5
-    - php: 5.6
-    - php: 7.0
-    - php: 7.1
     - php: 7.2
+    - php: 7.3
+    - php: 7.4
```
Comment (serbanghita) [inline .travis.yml:9]: We migrated to GH Actions yesterday, check https://github.com/serbanghita/Mobile-Detect/blob/master/.github/workflows/devel.yml
Can you include PHP 8 as well in the matrix?

File: .travis.yml [hunk 3]
```diff
@@ -4,14 +4,9 @@ language: php
...
     - php: nightly
   allow_failures:
     - php: nightly
```
Comment (serbanghita) [inline .travis.yml:12]: Do we want to test for edge?
I'd be more conservative (like the whole script) and put explicit versions.

**Review by serbanghita — CHANGES_REQUESTED** [review]

**Comment by serbanghita** [comment]
@bogdananton can you look into this PR as well?

**Comment by serbanghita** [comment]
@brainexe I will move as fast as I can. There is also this PR that I have to account for https://github.com/serbanghita/Mobile-Detect/pull/823
Can you also review the current PR changes?

**Comment by dereuromark** [comment]
Is there any update on this?

We have tons of those

> `PHP Deprecated:  preg_match(): Passing null to parameter #2 ($subject) of type string is deprecated in /home/runner/work/.../vendor/mobiledetect/mobiledetectlib/Mobile_Detect.php on line 1460`

in our CI system now for PHP 8.1+ etc.

**Comment by serbanghita** [comment]
I have opened a PR for support of PHP7 https://github.com/serbanghita/Mobile-Detect/pull/900

This PR contains shallow changes and I cannot continue with it in this state.
If any of you guys want to be a reviewer on https://github.com/serbanghita/Mobile-Detect/pull/900 - let me know

**Pull request closed without merging**